src/informo-feeder/poller/poller.go

```diff
@@ -28,143 +29,197 @@ import (
+	var err error
 
 	for {
-		timeToSleep, err := p.getDurationBeforePoll(feed)
+		// Load the last poll's results.
+		p.lastPollResults[feed.Identifier], err = p.db.GetItemsURLsForFeed(feed.Identifier)
 		if err != nil {
 			logrus.Panic(err)
 		}
 
-		time.Sleep(timeToSleep * time.Second)
+		logrus.WithFields(logrus.Fields{
+			"feed":  feed.Identifier,
+			"items": len(p.lastPollResults[feed.Identifier]),
+		}).Debug("Loaded last poll's results")
 
 		logrus.WithField("feedURL", feed.URL).Info("Polling")
 
-		_, latestItemTime, err := p.getLatestPosition(feed.URL)
-		if err != nil {
-			logrus.Panic(err)
-		}
-
+		// Retrieve the feed's XML.
 		resp, err := http.Get(feed.URL)
 		if err != nil {
 			logrus.Panic(err)
 		}
 
+		// If the server didn't reply with a 200 OK status code, wait for the
+		// correct amount of time, then jump to the next iteration.
 		if resp.StatusCode != http.StatusOK {
+			time.Sleep(time.Duration(feed.PollInterval) * time.Second)
+
 			continue
 		}
 
+		// Parse the XML retrieved from the remote server.
 		f, err := p.parser.Parse(resp.Body)
 		if err != nil {
 			logrus.Panic(err)
 		}
 
+		logrus.WithFields(logrus.Fields{
+			"feed":  feed.Identifier,
+			"items": len(f.Items),
+		}).Debug("Fetched feed")
+
+		// Iterate over the posts in chronological order. We can't promise to
+		// send all events chronologically (for example, if a new item appears
+		// in the middle of the feed between two iterations, we will send it
+		// after all the others, that we retrieved from the previous iteration),
+		// but we try to.
 		for i := len(f.Items) - 1; i >= 0; i-- {
 			item := f.Items[i]
-			if item.PublishedParsed.After(latestItemTime) {
-				var content string
-				if len(item.Content) > 0 {
-					content = item.Content
-				} else if len(item.Description) > 0 {
-					if htmlRegexp.MatchString(item.Description) {
-						content = item.Description
-					} else {
-						logrus.WithFields(logrus.Fields{
-							"title":         item.Title,
-							"publishedDate": item.PublishedParsed.String(),
-						}).Warn("Could not find any HTML content")
-
-						continue
-					}
-				}
-
-				logrus.WithFields(logrus.Fields{
-					"title":         item.Title,
-					"publishedDate": item.PublishedParsed.String(),
-				}).Info("Got a new item")
-
-				if err = p.replaceMedias(&content); err != nil {
+			// Only send the event if it wasn't part of the previous iteration.
+			if !p.itemKnown(feed.Identifier, item.Link) {
+				// Not findind any HTML in an item isn't a fatal error, log it
+				// and jump to the next iteration (after waiting enough).
+				if err = p.prepareThenSend(feed, item); err == errNoHTML {
+					logrus.WithFields(logrus.Fields{
+						"feed":          feed.Identifier,
+						"title":         item.Title,
+						"publishedDate": item.PublishedParsed.String(),
+					}).Warn("Could not find any HTML content")
+
+					time.Sleep(time.Duration(feed.PollInterval) * time.Second)
```

Comment:
Why wait here? This can cause huge sleep times if many items produce the errNoHTML error